src/view.ts

```diff
@@ -8,75 +8,121 @@ import type {
   Input,
   Output,
   Query,
-  ResultType,
+  ReadonlyJSONValue,
   Schema,
   TTL,
   ViewFactory,
 } from '@rocicorp/zero'
+import type { Ref } from 'vue'
 import { applyChange } from '@rocicorp/zero'
-import { reactive } from 'vue'
-
-interface QueryResultDetails {
-  readonly type: ResultType
+import { ref } from 'vue'
+
+// zero does not export this type
+type ErroredQuery = {
+  error: 'app'
+  queryName: string
+  details: ReadonlyJSONValue
+} | {
+  error: 'zero'
+  queryName: string
+  details: ReadonlyJSONValue
+} | {
+  error: 'http'
+  queryName: string
+  status: number
+  details: ReadonlyJSONValue
 }
 
-type State = [Entry, QueryResultDetails]
-
-const complete = { type: 'complete' } as const
-const unknown = { type: 'unknown' } as const
+export type QueryStatus = 'complete' | 'unknown' | 'error'
+
+export type QueryErrorDetails = {
+  type: 'app'
+  queryName: string
+  details: ReadonlyJSONValue
+} | {
+  type: 'http'
+  queryName: string
+  status: number
+  details: ReadonlyJSONValue
+}
 
 export class VueView<V> implements Output {
   readonly #input: Input
   readonly #format: Format
   readonly #onDestroy: () => void
   readonly #updateTTL: (ttl: TTL) => void
 
-  #state: State
+  #data: Ref<Entry>
+  #status: Ref<QueryStatus>
+  #error: Ref<QueryErrorDetails | undefined>
 
   constructor(
     input: Input,
     onTransactionCommit: (cb: () => void) => void,
-    format: Format = { singular: false, relationships: {} },
+    format: Format,
     onDestroy: () => void = () => {},
-    queryComplete: true | Promise<true>,
+    queryComplete: true | ErroredQuery | Promise<true>,
     updateTTL: (ttl: TTL) => void,
   ) {
     this.#input = input
     this.#format = format
     this.#onDestroy = onDestroy
     this.#updateTTL = updateTTL
-    this.#state = reactive([
-      { '': format.singular ? undefined : [] },
-      queryComplete === true ? complete : unknown,
-    ])
+    this.#data = ref({ '': format.singular ? undefined : [] })
+    this.#status = ref(queryComplete === true ? 'complete' : 'error' in queryComplete ? 'error' : 'unknown')
+    // @ts-expect-error - queryComplete is a weird union type causing stack depth limit error
+    this.#error = ref(queryComplete !== true && 'error' in queryComplete ? this.#makeError(queryComplete) : undefined)
```

Comment:
```suggestion
    this.#error = ref(queryComplete !== true && 'error' in queryComplete ? this.#makeError(queryComplete) : undefined) as Ref<QueryErrorDetails | undefined>
```

This would make the type-cast more explicit, and allows you to remove the expect-error directive (which will potentially hide other errors)